apps/src/lib/node/ledger/protocol/mod.rs

```diff
@@ -466,3 +531,65 @@ fn merge_vp_results(
         errors,
     })
 }
+
+#[cfg(test)]
+mod test {
+    use std::fs;
+    use std::path::PathBuf;
+
+    use serde_json::json;
+
+    use super::*;
+
+    // constructs a temporary fake wasm_dir with one wasm and a checksums.json
+    fn fake_wasm_dir(
+        wasm_name: impl AsRef<str>,
+        wasm_contents: impl AsRef<[u8]>,
+    ) -> PathBuf {
```

Comment:
There is probably a better place for this test helper, I took this from an old branch - https://github.dev/anoma/anoma/blob/f390706b6018e1ce46d76cc51c2d8671591b5925/tests/src/e2e/helpers/wasms.rs